# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2004/02/09 11:09:22-06:00 shaggy@... 
#   JFS: rename should update mtime on source and target directories
# 
diff -Nru a/fs/jfs/namei.c b/fs/jfs/namei.c
--- a/fs/jfs/namei.c	Tue Feb 10 15:33:40 2004
+++ b/fs/jfs/namei.c	Tue Feb 10 15:33:40 2004
@@ -1223,7 +1223,7 @@
 	old_ip->i_ctime = CURRENT_TIME;
 	mark_inode_dirty(old_ip);
 
-	new_dir->i_ctime = CURRENT_TIME;
+	new_dir->i_ctime = new_dir->i_mtime = CURRENT_TIME;
 	mark_inode_dirty(new_dir);
 
 	/* Build list of inodes modified by this transaction */
@@ -1235,7 +1235,7 @@
 
 	if (old_dir != new_dir) {
 		iplist[ipcount++] = new_dir;
-		old_dir->i_ctime = CURRENT_TIME;
+		old_dir->i_ctime = old_dir->i_mtime = CURRENT_TIME;
 		mark_inode_dirty(old_dir);
 	}
 
